internal/impl/update.go

```diff
@@ -90,21 +91,50 @@ func (d *Devbox) updateDevboxPackage(
 	if err != nil {
 		return err
 	}
-	if existing != nil && existing.Version != newEntry.Version {
+	if existing == nil {
+		ux.Finfo(d.writer, "Resolved %s to %[1]s %[2]s\n", pkg, newEntry.Resolved)
+		d.lockfile.Packages[pkg.Raw] = newEntry
+		return nil
+	}
+
+	if existing.Version != newEntry.Version {
 		ux.Finfo(d.writer, "Updating %s %s -> %s\n", pkg, existing.Version, newEntry.Version)
 		if err := d.removePackagesFromProfile(ctx, []string{pkg.Raw}); err != nil {
 			// Warn but continue. TODO(landau): ensurePackagesAreInstalled should
 			// sync the profile so we don't need to do this manually.
 			ux.Fwarning(d.writer, "Failed to remove %s from profile: %s\n", pkg, err)
 		}
 		d.lockfile.Packages[pkg.Raw] = newEntry
-	} else if existing == nil {
-		ux.Finfo(d.writer, "Resolved %s to %[1]s %[2]s\n", pkg, newEntry.Resolved)
-		d.lockfile.Packages[pkg.Raw] = newEntry
-	} else {
-		ux.Finfo(d.writer, "Already up-to-date %s %s\n", pkg, existing.Version)
+		return nil
+	}
+
+	// Check if the package's system info is missing, or not complete.
+	if featureflag.RemoveNixpkgs.Enabled() {
+		userSystem, err := nix.System()
+		if err != nil {
+			return err
+		}
+
+		// Check if the system info is missing for the user's system.
+		sysInfo := d.lockfile.Packages[pkg.Raw].Systems[userSystem]
+		if sysInfo == nil {
+			d.lockfile.Packages[pkg.Raw] = newEntry
```

Comment:
Wait, could this override an existing (different) system that already has a CA path?

Reply:
You are right! Much obliged 🙏🏾 

sending fix: https://github.com/jetpack-io/devbox/pull/1259